picard/ui/matchdetailsdialog/matchdetails.py

```diff
@@ -0,0 +1,211 @@
+
+        except TypeError as e:
+            log.warning("MATCH:No log entries for " + file_.filenamei + " due to " + e)
+
+        except Exception:
+            log.error(traceback.format_exc())
+
+    def accept_event(self, rows):
+        for row in rows:
+            self.load_selection(row)
+
+    def load_selection(self, row):
+        """Load the album corresponding to the selected track.
+
+        Allow user to pick from our new screen and use that track instead of one its currently assigned to.
+        Based on logic from searchdialog/track.py line 186
+        """
+
+        entry = self.entries[row]
+
+        if entry.rel_id != "":
+            # The track is not an NAT
+            self.tagger.get_release_group_by_id(entry.rg_id).loaded_albums.add(entry.rel_id)
+
+            # Have to move that file from its existing album to the new one.
+            if isinstance(self.current_file.parent, Track):
+                album = self.current_file.parent.album
+                self.tagger.move_file_to_track(self.current_file, entry.rel_id, entry.rec_id)
+                if album._files == 0:
+                    # Remove album if it has no more files associated
+                    self.tagger.remove_album(album)
+            else:
+                self.tagger.move_file_to_track(self.current_file, entry.rel_id, entry.rec_id)
+        else:
+            if self.current_file and getattr(self.current_file.parent, 'album', None):
+                album = self.current_file.parent.album
+                self.tagger.move_file_to_nat(self.current_file, entry.rec_id, None)
+                if album._files == 0:
+                    self.tagger.remove_album(album)
+            else:
+                self.tagger.load_nat(entry.rec_id, None)
+                self.tagger.move_file_to_nat(self.current_file, entry.rec_id, None)
+
+        log.debug("using alternative release id:" + entry.rel_id)
+
+    def export_match_dtl_entries_to_file(self):
+        """ Export entries to a file that can be viewed in external program
+        """
+        try:
+            # Prompt user for file location to save to
+            save_file = QFileDialog.getSaveFileName(None, 'Save File')
+            if save_file[0] == '':
+                return
+            else:
+                filename = save_file[0]
+
+            # Alternatively write to temp dir:
+            #filename = tempfile.gettempdir() + os.path.sep + "picard_match_details.tsv"
+
+            with open(filename, "w") as text_file:
+                print(match_details_log.match_dtl_entry.get_hdr_tsv(), file=text_file)
+
+                for row, entry in enumerate(self.entries):
+                    output_str = str(row) + "\t" + entry.fmt_as_tsv() + os.linesep
+                    text_file.write(output_str)
+            log.debug("Wrote file with match details to file:" + text_file.name)
+        except Exception:  # Ignore any errors as this output file is non-essential
```

Comment:
Which exceptions are supposed to be raised?

Reply:
Changed to IOError only.